qa/tasks/scripts.py

```diff
@@ -19,8 +19,17 @@ def __init__(self, ctx, logger):
             ctx['scripts_copied'] = True
 
     def run(self, remote, script_name, args=[], as_root=True):
-        self.log.info('(scripts) running script {} on remote {}'
-                      .format(script_name, remote.hostname)
+        remote_type = type(remote)
+        self.log.debug(
+            '(scripts) run method was passed a remote object of type {}'
+            .format(remote_type)
+            )
+        if remote_type == 'Cluster':
+            remote_spec = 'the whole cluster'
+        else:
+            remote_spec = 'remote {}'.format(remote.hostname)
```

Comment:
and then this conditional fails to recognize that this object is of class Cluster and the job fails lamely with `AttributeError: 'Cluster' object has no attribute 'hostname'`

(@smithfarm needs to learn object introspection in Python properly.)